drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/widgets/ScenarioGrid.java

```diff
@@ -246,8 +249,15 @@ protected boolean isPropertyAssigned(boolean instanceAssigned, FactMapping factM
      * @return
      */
     protected String getPlaceholder(boolean isPropertyAssigned, FactMapping factMapping) {
-        String editableCellPlaceholder = ScenarioSimulationUtils.getPlaceholder(factMapping.getClassName());
-        return isPropertyAssigned ? editableCellPlaceholder : ScenarioSimulationEditorConstants.INSTANCE.defineValidType();
+        if (isPropertyAssigned) {
+            if (ScenarioSimulationUtils.isExpressionType(factMapping)) {
+                return ScenarioSimulationEditorConstants.INSTANCE.insertExpression();
+            } else {
+                return ScenarioSimulationUtils.getPlaceholder(factMapping.getClassName());
+            }
+        } else {
+            return ScenarioSimulationEditorConstants.INSTANCE.defineValidType();
```

Comment:
Is it possible to refactor this method to move "everything" to `getPlaceholder`?